### PR TITLE
[runtime] add float for podvalue

### PR DESCRIPTION
**Pull request by antinucleon**
Thanks for contributing to TVM!   Please refer to guideline https://tvm.apache.org/docs/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

### CODE REVIEW

File: include/tvm/runtime/packed_func.h [hunk 1]
```diff
@@ -371,6 +371,20 @@ class TVMPODValue_ {
     TVM_CHECK_TYPE_CODE(type_code_, kDLFloat);
     return value_.v_float64;
   }
+  operator float() const {
+    double tmp_v = 0;
+    if (type_code_ == kDLInt) {
+      tmp_v = static_cast<double>(value_.v_int64);
+      CHECK_LE(tmp_v, std::numeric_limits<float>::max());
```
Comment (MarisaKirisame) [inline include/tvm/runtime/packed_func.h:378]: this three lines are redundent with the below 3 lines. can you have if else that set tmp_v's value?

Comment (antinucleon) [reply include/tvm/runtime/packed_func.h:378]: Depends whether we want to run TYPE_CHECK. If we want to run type check we have to have these redundant code.

File: include/tvm/runtime/packed_func.h [hunk 2]
```diff
@@ -371,6 +371,20 @@ class TVMPODValue_ {
...
+      CHECK_GE(tmp_v, std::numeric_limits<float>::min());
+      return static_cast<float>(tmp_v);
+    }
+    TVM_CHECK_TYPE_CODE(type_code_, kDLFloat);
+    tmp_v = static_cast<double>(value_.v_float64);
+    CHECK_LE(tmp_v, std::numeric_limits<float>::max());
+    CHECK_GE(tmp_v, std::numeric_limits<float>::min());
```
Comment (MarisaKirisame) [inline include/tvm/runtime/packed_func.h:385]:
```suggestion
    if (type_code_ == kDLInt) {
      tmp_v = static_cast<double>(value_.v_int64);
    } else {
      TVM_CHECK_TYPE_CODE(type_code_, kDLFloat);
      tmp_v = static_cast<double>(value_.v_float64);
    }
    CHECK_LE(tmp_v, std::numeric_limits<float>::max());
    CHECK_GE(tmp_v, std::numeric_limits<float>::min());
```

**Comment by tqchen** [comment]
Given that we already have operator double, let us simply ignore the float and use the double as positional argument in the function signature(say TypedPackedFunc, then convert later?)

File: include/tvm/runtime/packed_func.h (hunk 2 above)
Comment (MarisaKirisame) [reply include/tvm/runtime/packed_func.h:385]: Just to be clear I mean like this. I think it is still semantically equivalent.

**Comment by antinucleon** [comment]
@tqchen Sounds good.

**Pull request closed without merging**